### ISSUE TITLE
Very Annoying Pop-Up

### ISSUE THREAD

**Issue by CHuDWah** [antoyo/auto-sort-bookmarks#24]
Every time I open Firefox (v41.0.1), I get the start page and a second window with the following.  Checking "Do not show again" has no effect.  VERY ANNOYING!!

Auto-Sort Bookmarks

I am looking for someone to maintain Auto-Sort Bookmarks.

If you are a JavaScript developer and want to contribute to this extension so that it stays compatible with Firefox, please open an issue on Github.
Do not show again

**Comment by TheRealMarnes** [comment]
Do you close the popup by (checking the checkbox and) clicking the close button below the checkbox, or do you just close the window?

**Comment by jmd13391** [comment]
FYI... The "Close" button does not close the window. The "Do not show again" checkbox appears to have no affect. I agree with CHuDWah... the pop-up is extremely annoying. Guess it's time to look for a new FF Bookmark sorter. It's a shame... Auto-Sort Bookmarks rocked up until now.

**Comment by TheRealMarnes** [comment]
@jmd13391 @CHuDWah FYI it works fine for me and doesn't show up after the first time. You guys are on the latest firefox version and whatnot?

Go into about:config and set extensions.sortbookmarks@bouanto.do_not_show_contributor_dialog to true to stop the popup.

And don't be so spiteful :) It's not like there is another bookmark sorter out there anyway. This one will be maintained at some point, possibly by me and otherwise someone else.

**Comment by TheRealMarnes** [comment]
Do you guys by any chance have javascript disabled?

**Comment by derwaldgeist** [comment]
I can confirm the dialog to work as expected (MacOS, FF 41.0.1). Just dropped by because I was curious if anybody could be found for maintaining this great plugin. Thanks to the maintainers for their great work! And to the complainers above: Calm down guys. It happens once in a while that code is buggy (if it even is). If you don't like it anymore, just go for an alternative - or program one yourself... There's always the plugin removal button at your service. Nobody is forcing you to use free software. This doesn't fall from heavens.

**Comment by jmd13391** [comment]
Hi TheRealMarnes, For the record, I am not "complaining". I was simply attempting to provide you with factual information about the button and checkbox functionality for me. It was not my intent to be "overly negative". I would like to think that I can offer an opinion (annoying) without being tagged a "complainer". Just FYI: Javascript is not disabled. Latest version of FF (41.0.1) is in use. Thank you for the "go into about:config..." suggestion but I have no intentions of manually altering my FF configuration to workaround the button and checkbox options not working. I like the plug-in, I think it does an excellent job at filling a need (sorting bookmarks) and appreciate the effort that went into coding it. I also fully understand the intent of the pop-up. I simply wish that it could stay disabled and/or closed when I select those options on the page. Have a great day and good luck with your call for assistance.  :-)

**Comment by antoyo** [comment]
@jmd13391 For your information, checking the checkbox is supposed to set `extensions.sortbookmarks@bouanto.do_not_show_contributor_dialog` to `true`, so doing it manually will do the same and you won't see this page again.

Do you see any errors in the browser console (or in this page developer console) after (or before) clicking on a button?

Anyway, this page will probably be removed soon.

**Comment by Gitoffthelawn** [comment]
I just tested it, and it worked fine for me.  From the VERY ANNOYING posts above, sounds like it is not working for at least two people though.  Just stating my opinion ;-)

**Comment by miketaylr** [comment]
FWIW, this is confusing your users. 

https://github.com/webcompat/web-bugs/issues/1810

**Comment by TheRealMarnes** [comment]
@miketaylr that is pretty exaggerated. If he hadn't put in that popup, I would never have known the addon was being discontinued either. How do people expect devs to take care of their program and get messages to users if users don't read release notes, don't have to visit the addon page, and now don't even want a simple one-time message displayed to them? Would you rather subscribe to a mailing list? oh wait users won't do that either.

Also, how is the message popup confusing or in any other way a bad thing?
"Auto-sort bookmarks needs new developers"
"<don't show again> OK"
hurr durr this is a serious problem, uninstall that addon

This whole reaction goes hand in hand with what @jmd13391 was saying earlier: "I'm not gonna do that and manually start changing my ff config to fix this addon's popup". That statement shows that not only does he not know the basics of firefox (i.e. that older addon prefs are stored in about:config, which was standard until recently, and very obvious from the name of the pref), he also maybe shouldn't even be using firefox if he's against using about:config. That's like using linux when you're phobic of CLI.

Half of what's being said in this thread sounds like what you hear when avid windows users try linux and can't deal with the fact that linux is not windows, and blame linux for that, you know?

**Comment by miketaylr** [comment]
> that is pretty exaggerated

What is exaggerated? I wrote "this is confusing your users."

If you follow the link, it's pretty clear that person has no idea what is happening.

> Is this a Firefox issue or a Google issue, and how does one get rid of it?

**Comment by TheRealMarnes** [comment]
> Is this a Firefox issue or a Google issue, and how does one get rid of it?
That part is exaggerated on the user's part. I mean, how the heck do they miss that? It's literally the title of the page.

What I meant was exaggerated on your part was the whole
> But the reason this is happening is because you've installed "Auto-Sort Bookmarks" add-on and apparently the author thought this was a good idea. Personally, I would uninstall the add-on because this pretty bogus.
thing.

**Comment by miketaylr** [comment]
> I mean, how the heck do they miss that? It's literally the title of the page.

People can be very surprising. ^_^

> What I meant was exaggerated on your part was the whole

No, that's not exaggerated on my part. It's honestly how I would react. I would uninstall it. Especially if the "don't show again" link didn't work (as some users are reporting here).

 But I'm not a user of the add-on, so none of this matters. I'm just linking the issues because they're related.

**Comment by TheRealMarnes** [comment]
>But I'm not a user of the add-on, so none of this matters. I'm just linking the issues because they're related.
"Don't shoot the messenger" true, sorry.

I just wish people would stop making such a storm about something that isn't as big as issue as they're making it out to be. And yes, it seems the "do not show again" is bugged, but when I look at the source code I see no potential for errors whatsoever so I believe they must have done something with their setup (apparently not disabling js, but something equally disruptive) to cause it on their end.

**Comment by antoyo** [comment]
For your information, I posted in some forums more than one month ago without getting much responses.
One person suggested to show a message to the users.
I thought it was a bit too aggressive, but I did not want to let break an add-on used by more than 19 000 users, so I decided to update the add-on to show the message.

Anyway, perhaps it would have been better to show it only once.

**Comment by CHuDWah** [comment]
@TheRealMarines @derwaldgeist Lose the butthurt.  I reported a bug like you guys encourage users to do.  I didn’t object to the popup recruiting JavaScript developers.  If I was one, I might have volunteered.  But I’m not, so I didn’t.  And in bitching about what users can’t/won’t do, you seem to have lost sight of the fact that most of them aren’t developers either.  Just because I know how to drive a car doesn’t mean I do, or should have to, know how to rebuild the engine.  And I’d be scared to try for fear of screwing it up.  Yet you immediately jumped to the conclusion that screwing it up was exactly what I did.

What I objected to was that checking the box to make the popup go away didn’t work.  I said getting, and having to close, the popup every time I opened Firefox is VERY ANNOYING because it is.  If you think that’s “overly negative”, you need to grow thicker skin.  Oh, and you asked if I am using the latest Firefox version, if I checked the “Do not show again” box, etc.  In hopes of facilitating a diagnosis of the problem, I purposely included all that in my original post.  Reading comprehension is fundamental.  Learn to do it.

Now, for your information, setting 

extensions.sortbookmarks@bouanto.do_not_show_contributor_dialog

to true fixed the problem.  But here’s the thing, that preference did not appear in my about:config.  I’m knowledgeable (and brave) enough to add it.  But had I not posted the bug and received replies about the preference, I’d have no clue as to how to fix it.

Auto-sort is an excellent add-on, even better because it’s free.  I've never had a problem with it before and maybe that's what made this one even more frustrating.  I might be more lenient with an add-on I expected to be buggy, although I probably would just uninstall it and be done with it.  I have no intention of doing that with auto-sort.  I have lots of bookmarks and managing them would be difficult without the add-on.  I do appreciate the hard work of those who authored and maintain it, and hope they continue to do so.  But that doesn’t mean I’m not going to report (and yes, complain about) a bug.  So deal with it.

**Comment by foliosi** [comment]
I'm using Firefox 41.0.2 on WinXP SP03 and i'm experiencing exactly the same problem reported by CHuDWah (i repeatedly tried to check the box to make the popup go away, but it didn’t work and the registry extensions.sortbookmarks@bouanto.do_not_show_contributor_dialog does not appear in about:config).
Could help us ?
Many thanks in advance.
Federico Oliosi

**Comment by TheRealMarnes** [comment]
@foliosi you can just create the entry if it doesn't exist (right-click, new, boolean).

Again, could someone for whom the checkbox does not work please give us some console output? Does your web/dev/js console (ctrl-shift-k or ctrl-shift-j) show any errors when the popup opens or when you tick the checkbox? Do you have any addons or other things in place that block javascript or changes to your config?

--

@CHuDWah you're mostly right, we do want you to report bugs and such, that's a good thing. It's just all the complaining besides the bug (not just from you) that's going on that's getting to me. Nobody has any use for snide remarks and such in a bug report, things like "VERY ANNOYING!" are stating the obvious and make it sound as if you think anyone designed it to work that way.

Also, using about:config is a standard thing to know in firefox. You're right, you don't need to know how to assemble an engine, but if you're gonna use firefox you should know firefox basic usage and that includes about:config. You don't hop in a car and start driving it without learning how to drive first and then expect other drivers not to honk at you.

And yes, I read about the latest version in your original post the first time, I forgot and skimmed over it after that. It's not like your bug is the center of my life atm, I have fulltime work and things to do, and snide remarks like that reading comprehension are again uncalled for.

>But had I not posted the bug and received replies about the preference, I’d have no clue as to how to fix it.

Yes, that's kind of expected. Again you're implying someone designed it to work that way (to not work). Users weren't supposed to have to mess with prefs to disable the poup, something else is going wrong at your end.

>But that doesn’t mean I’m not going to report (and yes, complain about) a bug. So deal with it.

Nobody asked you not to report bugs. We do just want you to report them in a neutral way, that's all.

Can we drop this now please? We know about this bug and someone will look into it when we can. Leave any useful information you have about what might be causing it for you and otherwise leave this issue (topic/thread) alone.

**Comment by TheRealMarnes** [comment]
Btw if you already added the extensions.sortbookmarks@bouanto.do_not_show_contributor_dialog pref, you can just remove it or set it to false to make the popup come back, so you can check if it causes any errors in the console for us. Some console output and other info as per my previous post would be much appreciated because at the moment there aren't any leads about the cause.

**Comment by foliosi** [comment]
I made some experiments:
- the problem affects different Windows versions (Seven and XP);
- the problem remains unsolved with or without java blocker enabled;
- the problem can be manually fixed by adding the new entry extensions.sortbookmarks@bouanto.do_not_show_contributor_dialog in about:config file.

The web/dev/js console is as follows:
mutating the [[Prototype]] of an object will cause your code to run very slowly; instead create the object with the correct initial [[Prototype]] value using Object.create frameworks-f8473dece7242da6a20d52313634881b3975c52cebaa1e6c38157c0f26185691.js:1:26491

Many thanks for your attention.

**Comment by TheRealMarnes** [comment]
@foliosi thanks, but unfortunately there's no clue in there. Also, fyi, java is not javascript. The error is from somewhere else too probably.

**Comment by foliosi** [comment]
I'm sorry, but i'm not a code-writer nor a programmer.
Below you find the FF console messages transcription before and after clicking the auto-sort-bookmark checkbox.
Many thanks again for your attention (and patience)
---
CTRL+SHIFT+K AFTER OPENING FF 41.0.1 BUT BEFORE CLICKING THE AUTO-SORT-BOOKMARK CHECKBOX

Could not read chrome manifest 'file:///C:/Programmi/Mozilla%20Firefox/chrome.manifest'.
Get a connection to permissions.sqlite.
Il “Content-Type“ HTTP “video/mp4” non è supportato. Impossibile completare il caricamento della risorsa multimediale chrome://gfxsanity/content/videotest.mp4. sanitytest.html
DEPRECATION WARNING: FUEL is deprecated, you should use the add-on SDK instead.
You may find more details about this deprecation at: https://developer.mozilla.org/Add-ons/SDK/
jar:file:///C:/Programmi/Mozilla%20Firefox/browser/omni.ja!/components/fuelApplication.js 1458 Application
jar:file:///C:/Programmi/Mozilla%20Firefox/browser/omni.ja!/components/fuelApplication.js 726 af_ci
chrome://web2pdfextension/content/browser_status.js 65 null
null 0 null
 Deprecated.jsm:79:0
unreachable code after return statement elements.js:987:2
unreachable code after return statement elements.js:998:2
unreachable code after return statement elements.js:1012:2
Proprietà sconosciuta “-moz-border-radius”.  Dichiarazione tralasciata. tftoolbar.css:183:19
DEPRECATION WARNING: nsIFormHistory2 is deprecated and will be removed in a future version
You may find more details about this deprecation at: https://bugzilla.mozilla.org/show_bug.cgi?id=879118
resource://gre/components/nsFormHistory.js 20 FormHistory
resource://gre/modules/XPCOMUtils.jsm 324 XPCOMUtils__getFactory/factory.createInstance
chrome://tftoolbar/content/tftoolbar.js 2 null
null 0 null
 Deprecated.jsm:79:0
Proprietà sconosciuta “-moz-border-radius”.  Dichiarazione tralasciata. foxmarks.css:187:19
Proprietà sconosciuta “-moz-border-radius”.  Dichiarazione tralasciata. foxmarks.css:203:19
Selettore previsto.  Insieme di regole ignorato a causa del selettore errato. lastpass.css:16:55
Errore durante l’interpretazione di “background”.  Dichiarazione tralasciata. lastpass.css:18:63
Proprietà sconosciuta “-moz-border-radius”.  Dichiarazione tralasciata. lastpass.css:18:208
Errore durante l’interpretazione di “background”.  Dichiarazione tralasciata. lastpass.css:18:313
Errore durante l’interpretazione di “display”.  Dichiarazione tralasciata. browser.xul
gBrowser.addProgressListener was called with a second argument, which is not supported. See bug 608628. Call stack: addProgressListener@chrome://browser/content/tabbrowser.xml:2610:69
TFTB_Init@chrome://tftoolbar/content/tftoolbar.js:269:3
 tabbrowser.xml:2607:0
No chrome package registered for chrome://sessionmanager/content/modules/session_manager.jsm
gBrowser.addProgressListener was called with a second argument, which is not supported. See bug 608628. Call stack: addProgressListener@chrome://browser/content/tabbrowser.xml:2610:69
TFTB_Init@chrome://tftoolbar/content/tftoolbar.js:269:3
 tabbrowser.xml:2607:0
Evento da tastiera non disponibile in alcuni layout di tastiera: tasto=“h” modificatori=“control alt” browser.xul
Evento da tastiera non disponibile in alcuni layout di tastiera: tasto=“n” modificatori=“accel, alt” browser.xul
Evento da tastiera non disponibile in alcuni layout di tastiera: tasto=“t” modificatori=“accel,alt” browser.xul
Evento da tastiera non disponibile in alcuni layout di tastiera: tasto=“r” modificatori=“accel,alt” browser.xul
Evento da tastiera non disponibile in alcuni layout di tastiera: tasto=“c” modificatori=“accel,alt” browser.xul
Evento da tastiera non disponibile in alcuni layout di tastiera: tasto=“i” modificatori=“accel,alt,shift” browser.xul
GET 
XHR 
https://aus4.mozilla.org/update/3/GMP/41.0.2/20151014143721/WINNT_x86-msvc/en-US/release/Windows_NT%205.1.3.0%20(x86)/default/default/update.xml [HTTP/1.1 200 OK 735ms]
Questo sito utilizza un certificato SHA-1. Si consiglia di utilizzare certificati basati su algoritmi di firma con funzioni hash più robuste di SHA-1.[Ulteriori informazioni] aus4.mozilla.org


CTRL+SHIFT+K AFTER OPENING FF 41.0.1 AND AFTER CLICKING THE AUTO-SORT-BOOKMARK CHECKBOX

Could not read chrome manifest 'file:///C:/Programmi/Mozilla%20Firefox/chrome.manifest'.
Get a connection to permissions.sqlite.
Il “Content-Type“ HTTP “video/mp4” non è supportato. Impossibile completare il caricamento della risorsa multimediale chrome://gfxsanity/content/videotest.mp4. sanitytest.html
DEPRECATION WARNING: FUEL is deprecated, you should use the add-on SDK instead.
You may find more details about this deprecation at: https://developer.mozilla.org/Add-ons/SDK/
jar:file:///C:/Programmi/Mozilla%20Firefox/browser/omni.ja!/components/fuelApplication.js 1458 Application
jar:file:///C:/Programmi/Mozilla%20Firefox/browser/omni.ja!/components/fuelApplication.js 726 af_ci
chrome://web2pdfextension/content/browser_status.js 65 null
null 0 null
 Deprecated.jsm:79:0
unreachable code after return statement elements.js:987:2
unreachable code after return statement elements.js:998:2
unreachable code after return statement elements.js:1012:2
Proprietà sconosciuta “-moz-border-radius”.  Dichiarazione tralasciata. tftoolbar.css:183:19
DEPRECATION WARNING: nsIFormHistory2 is deprecated and will be removed in a future version
You may find more details about this deprecation at: https://bugzilla.mozilla.org/show_bug.cgi?id=879118
resource://gre/components/nsFormHistory.js 20 FormHistory
resource://gre/modules/XPCOMUtils.jsm 324 XPCOMUtils__getFactory/factory.createInstance
chrome://tftoolbar/content/tftoolbar.js 2 null
null 0 null
 Deprecated.jsm:79:0
Proprietà sconosciuta “-moz-border-radius”.  Dichiarazione tralasciata. foxmarks.css:187:19
Proprietà sconosciuta “-moz-border-radius”.  Dichiarazione tralasciata. foxmarks.css:203:19
Selettore previsto.  Insieme di regole ignorato a causa del selettore errato. lastpass.css:16:55
Errore durante l’interpretazione di “background”.  Dichiarazione tralasciata. lastpass.css:18:63
Proprietà sconosciuta “-moz-border-radius”.  Dichiarazione tralasciata. lastpass.css:18:208
Errore durante l’interpretazione di “background”.  Dichiarazione tralasciata. lastpass.css:18:313
Errore durante l’interpretazione di “display”.  Dichiarazione tralasciata. browser.xul
gBrowser.addProgressListener was called with a second argument, which is not supported. See bug 608628. Call stack: addProgressListener@chrome://browser/content/tabbrowser.xml:2610:69
TFTB_Init@chrome://tftoolbar/content/tftoolbar.js:269:3
 tabbrowser.xml:2607:0
No chrome package registered for chrome://sessionmanager/content/modules/session_manager.jsm
gBrowser.addProgressListener was called with a second argument, which is not supported. See bug 608628. Call stack: addProgressListener@chrome://browser/content/tabbrowser.xml:2610:69
TFTB_Init@chrome://tftoolbar/content/tftoolbar.js:269:3
 tabbrowser.xml:2607:0
Evento da tastiera non disponibile in alcuni layout di tastiera: tasto=“h” modificatori=“control alt” browser.xul
Evento da tastiera non disponibile in alcuni layout di tastiera: tasto=“n” modificatori=“accel, alt” browser.xul
Evento da tastiera non disponibile in alcuni layout di tastiera: tasto=“t” modificatori=“accel,alt” browser.xul
Evento da tastiera non disponibile in alcuni layout di tastiera: tasto=“r” modificatori=“accel,alt” browser.xul
Evento da tastiera non disponibile in alcuni layout di tastiera: tasto=“c” modificatori=“accel,alt” browser.xul
Evento da tastiera non disponibile in alcuni layout di tastiera: tasto=“i” modificatori=“accel,alt,shift” browser.xul
Questo sito utilizza un certificato SHA-1. Si consiglia di utilizzare certificati basati su algoritmi di firma con funzioni hash più robuste di SHA-1.[Ulteriori informazioni] aus4.mozilla.org
1445065296986	addons.update-checker	WARN	onUpdateCheckComplete notification failed: [Exception... "Component returned failure code: 0x804b000a (NS_ERROR_MALFORMED_URI) [nsIIOService.newURI]"  nsresult: "0x804b000a (NS_ERROR_MALFORMED_URI)"  location: "JS frame :: resource://gre/modules/NetUtil.jsm :: NetUtil_newURI :: line 191"  data: no] Stack trace: NetUtil_newURI()@resource://gre/modules/NetUtil.jsm:191 < AI_createUpdate()@resource://gre/modules/addons/XPIProvider.jsm:6191 < UC_onUpdateCheckComplete()@resource://gre/modules/addons/XPIProvider.jsm:6431 < UP_onLoad()@resource://gre/modules/addons/AddonUpdateChecker.jsm:498 < loadEventListener()@resource://gre/modules/addons/AddonUpdateChecker.jsm:422 < <file:unknown>
1445065297428	addons.update-checker	WARN	Update manifest for web2pdfextension@web2pdf.adobedotcom did not contain an updates property
1445065297672	addons.update-checker	WARN	Update manifest for {972ce4c6-7e08-4474-a285-3208198ce6fd} did not contain an updates property
Evento da tastiera non disponibile in GTK2: tasto=“u” modificatori=“shift, accel” toolbox.xul

**Comment by duluthga** [comment]
I too am getting these extremely annoying pop ups. This group appears to be comprised of very bright computer gurus but i'm just a regular computer user. PLEASE, in terms I can understand, how do I get rid of this annoying pop up???? HELP!

**Comment by TheRealMarnes** [comment]
@duluthga 
>Go into about:config and set extensions.sortbookmarks@bouanto.do_not_show_contributor_dialog to true to stop the popup.

http://lmgtfy.com/?q=how+to+use+firefox+about%3Aconfig

**Comment by antoyo** [comment]
@foliosi The only error I see in these logs is not related to this add-on. It seems to come from the Firefox Add-on Update Checker.

@TheRealMarnes Unless someone wants to investigate more, I think the best solution is to update the add-on to only show the dialog once (removing the need for the user to check the checkbox).

**Comment by TheRealMarnes** [comment]
@antoyo I'll get on it then

**Comment by antoyo** [comment]
Thanks.
I dislike to be unable to reproduce other users' issues, but it seems to happen from time to time with Firefox add-ons and I don't know why.

**Comment by TheRealMarnes** [comment]
@antoyo Yeah this particular bug is pretty nasty, and now that I'm working on it (which is a dead-simple fix) I'm getting pretty strange and unusual behavior. There's something deeper wrong here, like how the paperwork in package.json was a little out of format.

**Comment by TheRealMarnes** [comment]
@antoyo so the problems boils down to

```javascript
ActionButton({
        id: 'auto-sort-bookmarks-sort-all-toolbar',
        icon: {
            16: './icon-16.png',
            32: './icon-32.png',
            64: './icon-64.png',
        },
        label: _('sort_bookmarks'),
        onClick: sortAllBookmarks
    });
```
sortbookmarks:Plural form unknown for locale "null"
sortbookmarks:TypeError: aId is undefined (referring to actionButton id)

**Comment by TheRealMarnes** [comment]
Well, a new maintenance update is awaiting moderation now (won't take long), and that should fix the popup issue. You'll only get to see it once.

**Comment by antoyo** [comment]
@TheRealMarnes Do you still need an answer to the question in your previous post?
For now, I don't know why the id error happens.
As for the locale warning, I saw it a lot of times, but I don't know why it happens and it does not seem to cause any issue (it would be great to fix it though).
Thanks for your work on this issue.

**Comment by duluthga** [comment]
Thanks for working on this guys. I have no idea how I got it but it has to be affecting thousands of people.

**Comment by antoyo** [comment]
@CHuDWah, @jmd13391, @foliosi, @duluthga please confirm that the new version of Auto-Sort Bookmarks works as expected so that we can close this issue.

**Comment by duluthga** [comment]
No joy. Pop up still activates prior to browser opening. ‘Do No Show Again’ check box does not work.

**Comment by TheRealMarnes** [comment]
@duluthga if you're still seeing a "do not show again" checkbox, you're not on the current version. Either that or the page got cached and you're getting the cached version, which would be a bit odd (and in which case you should clear your cache and try again)...

**Comment by duluthga** [comment]
Have cleared cache, did cold reboot, still there. Ran Norton deep scan as well.

**Comment by antoyo** [comment]
Sometimes, I takes some time for Firefox to update the add-on.
Makes sure you have the version 2.10.4 (in the add-on manager).
If not, you can do a manual update in a Firefox menu of the add-on manager.

Of course, after updating to the new add-on, you will see the page once, but after that, you should not see it again.

**Comment by duluthga** [comment]
Bingo! Looks like it’s gone. I drive a Boeing 777 for a living…computers, it’s a love/hate relationship! Thanks!!

**Comment by Gitoffthelawn** [comment]
*I drive a Boeing 777 for a living…computers, it’s a love/hate relationship!*

@duluthga That's one the most disturbing things I've heard in a long time.  The 777 cockpit is a glass cockpit.  It is all computers.  In addition, you have no physical connections to your flight surfaces; all your connections are established by onboard computers.

**Comment by duluthga** [comment]
Yep, almost 3000 hours, amazing piece of equipment.
